### PR TITLE
fix: [CDS-69576]: Remove "Java" logs when Yaml processing fails

### PR DESCRIPTION
**Pull request by alebuh**
## Describe your changes
Simplifying logs by removing
 "Java" parts from it
If Secret yaml is invalid output will look like this:
Screenshot 2023-05-23 at 11 21 22
If any resource is any other kind it will print a more detailed explanation:
Screenshot 2023-05-23 at 11 15 19

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48112)

### CODE REVIEW

**Comment by alebuh** [comment]
trigger uts

**Review by abhinavsingh1196 — DISMISSED** [review]

File: 970-api-services-beans/src/test/java/io/harness/k8s/model/KubernetesResourceTest.java [hunk 1]
```diff
@@ -897,8 +898,48 @@ public void testYamlConstructorMessageExtraction() throws IOException {
...
+  @Test
+  @Owner(developers = BUHA)
+  @Category(UnitTests.class)
+  public void testYamlConstructorMessageExtractionWithDeploymentErrors() throws IOException {
+    URL url = this.getClass().getResource("/deployment-invalid-value.yaml");
+    String fileContents = Resources.toString(url, StandardCharsets.UTF_8);
+    String expectedError = "Failed to load spec for resource kind: Deployment, name: myapp-deployment-2 \n"
+        + "Cannot create spec for V1Deployment  \n"
+        + "  line 10, column 3:\n"
+        + "      template:\n"
```
Comment (abosii-harness) [inline 970-api-services-beans/src/test/java/io/harness/k8s/model/KubernetesResourceTest.java:914]: I believe we should show the full stack, as this error doesn't show the exact what is wrong here

**Review by abosii-harness — CHANGES_REQUESTED** [review]

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/KubernetesResource.java [hunk 2]
```diff
@@ -612,17 +612,19 @@ public Object getK8sResource() {
       }
     } catch (Exception ex) {
       String yamlConstructionExceptionMessage =
-          ex instanceof ConstructorException ? extractMessageHeaderWithProblemMarks(ex) : "";
+          ex instanceof ConstructorException && !"Secret".equals(this.resourceId.getKind())
```
Comment (abosii-harness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/model/KubernetesResource.java:615]: equalsIgnoreCase just in case

Comment (alebuh) [reply 970-api-services-beans/src/main/java/io/harness/k8s/model/KubernetesResource.java]: done

File: 970-api-services-beans/src/test/java/io/harness/k8s/model/KubernetesResourceTest.java (hunk 1 above)
Comment (alebuh) [reply 970-api-services-beans/src/test/java/io/harness/k8s/model/KubernetesResourceTest.java:914]: done

**Comment by alebuh** [comment]
trigger uts

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by alebuh** [comment]
trigger ut5

**Review by abosii-harness — APPROVED** [review]

**Comment by acasian-harness** [comment]
@alebuh do we need to consider the same for configMap?

**Review by acasian-harness — DISMISSED** [review]

File: 970-api-services-beans/src/main/java/io/harness/k8s/model/KubernetesResource.java [hunk 3]
```diff
@@ -612,19 +610,33 @@ public Object getK8sResource() {
       }
     } catch (Exception ex) {
       String yamlConstructionExceptionMessage =
-          ex instanceof ConstructorException ? extractMessageHeaderWithProblemMarks(ex) : "";
+          ex instanceof ConstructorException && !"Secret".equalsIgnoreCase(this.resourceId.getKind())
```
Comment (acasian-harness) [inline 970-api-services-beans/src/main/java/io/harness/k8s/model/KubernetesResource.java:613]: nit: we have `io.harness.k8s.model.Secret` kind constant

**Comment by alebuh** [comment]
trigger uts

**Review by abosii-harness — APPROVED** [review]

**Comment by alebuh** [comment]
trigger uts

**Comment by harshjain12** [comment]
trigger ut0

**Review by abhinavsingh1196 — APPROVED** [review]

**Comment by alebuh** [comment]
trigger review

**Comment by alebuh** [comment]
trigger review

**Pull request merged**